os/kernel/binary_manager/binary_manager.c

```diff
@@ -143,14 +157,25 @@ int binary_manager(int argc, char *argv[])
 		case BINMGR_GET_INFO_ALL:
 			binary_manager_get_info_all(request_msg.requester_pid);
 			break;
-		case BINMGR_GET_STATE:
-			binary_manager_get_state_with_name(request_msg.requester_pid, (char *)request_msg.data.bin_name);
+		case BINMGR_CREATE_BIN:
+			binary_manager_create_entry(request_msg.requester_pid, request_msg.data.update_bin.bin_name, request_msg.data.update_bin.version);
 			break;
 		case BINMGR_UPDATE:
+			if (!strncmp("kernel", request_msg.data.bin_name, BIN_NAME_MAX)) {
```

Comment:
In flat build, do we need to check this name?
If it is, then you need to add failure case. I mean that binary name is not "kernel".
If it isn't, then you can move this conditional into CONFIG_APP_BINARY_SEPARATION.

Reply:
Yes. I think we need to check name for invalid string on flat build.
I agree with your opinion but currently it doesn't return a result. I will think more.